pgxn/neon/communicator.c

```diff
@@ -2577,3 +2702,19 @@ communicator_processinterrupts(void)
 
 	return prev_interrupt_cb();
 }
+
+PG_FUNCTION_INFO_V1(neon_communicator_min_inflight_request_lsn);
+
+Datum
+neon_communicator_min_inflight_request_lsn(PG_FUNCTION_ARGS)
+{
+	XLogRecPtr min_lsn = RecoveryInProgress()
+		? GetXLogReplayRecPtr(NULL)
+		: InfiniteXLogRecPtr;
```

Comment:
Why is this function behaving different in replicas than primary?
IMO the tracking of min inflight request LSN that we're adding here should be a freestanding concept.
It shouldn't care for what it is being used.